File: packages/lib-solid/src/avatars/avatar-pixel.utils.ts
import { hashCode, getRandomColor } from "../avatar.utils";

export const ELEMENTS = 64;
export const SIZE = 80;

export function generateColors(name: string, colors: string[]) {
  const numFromName = hashCode(name);
  const range = colors && colors.length;

  const colorList = Array.from({ length: ELEMENTS }, (_, i) =>
    getRandomColor(numFromName % i, colors, range)
  );

  return colorList;
}

File: packages/lib-leptos/src/components/avatar_pixel_utils.rs
use crate::components::avatar_utils::*;

pub const SIZE: usize = 80;
pub const ELEMENTS: usize = 64;

#[derive(Debug, Clone)]
pub struct GeneratedColor {
    pub index: usize,
    pub color: String,
}

pub fn generate_colors(name: &str, colors: Vec<&str>) -> Vec<GeneratedColor> {
    let num_from_name = hash_code(name);
    let range = colors.len();

    let mut generated_colors: Vec<GeneratedColor> = vec![];

    for i in 0..ELEMENTS {
        generated_colors.push(GeneratedColor {
            color: get_random_color(num_from_name + i as i64, &colors, range as i64),
            index: i,
        });
    }

    generated_colors
}
